src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+    private int halfOfSize = size / 2;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node currentNode = new Node<>(null, value, null);
+            tail = currentNode;
+            head = currentNode;
+            size++;
+        } else if (size > 0) {
+            Node previousNode = head;
+            Node currentNode = new Node(previousNode, value, null);
+            previousNode.next = currentNode;
+            head = currentNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Your index was: " + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0 && size > 0) {
```

Comment:
You've already checked for `index == size` so in this case size will always be greater than 0 - simplify the condition.